### PR TITLE
Jwt implementation

### PR DESCRIPTION
**Pull request by billzyc**

### CODE REVIEW

**Comment by billzyc** [comment]
whoops

File: flask/app/views.py [hunk 1]
```diff
@@ -48,6 +59,43 @@ def authorize():
     return redirect('/')
 
 
+@app.route('/googleSignin', methods=['POST'])
+def googleSignin():
+    try:
+        google_token = request.json['gToken']
+        id_info = id_token.verify_oauth2_token(
+            google_token, requests.Request(), os.environ.get('GOOGLE_CLIENT_ID', None))
```
Comment (billzyc) [inline flask/app/views.py:67]: this

File: react-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { CookiesProvider } from 'react-cookie';
+import './App.scss';
+import Login from './pages/login/login.js';
+
+function App() {
+
+
+  return (
+    <CookiesProvider>
+      <Login/>
```
Comment (billzyc) [inline react-app/src/App.js:10]: component here just temporary for testing purposes. Fixed in the react-app pr

File: react-app/src/pages/login/login.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { CookiesProvider } from 'react-cookie';
+import './login.scss';
+import GoogleOAuth from '../../components/googleOAuth/googleOAuth.js';
+
+function Login() {
+
+
+  return (
+    <CookiesProvider>
```
Comment (billzyc) [inline react-app/src/pages/login/login.js:9]: CookiesProvider should not be here. Removed in react-app branch

File: react-app/src/styles/fonts.scss [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// Fonts
```
Comment (billzyc) [inline react-app/src/styles/fonts.scss:1]: File removed in other branch

File: react-app/src/components/googleOAuth/googleOAuth.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import GoogleLogin from "react-google-login";
```
Comment (billzyc) [inline react-app/src/components/googleOAuth/googleOAuth.js:1]: fixed the casing for all components in other branch. E.g googleOAuth -> GoogleOAuth

File: react-app/src/components/googleOAuth/googleOAuth.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      "http://127.0.0.1:5001/googleSignin",
+      requestOptions
+    );
+
+    const data = await apiResponse.json();
+    setCookie('token',data.token, { path: '/' });
+  };
+
+  const onFail = () => {
+    console.log("failed!");
```
Comment (billzyc) [inline react-app/src/components/googleOAuth/googleOAuth.js:30]: placeholder

File: react-app/src/styles/mixins.scss [hunk 7]
```diff
@@ -0,0 +1,139 @@
+@import 'vars.scss';
```
Comment (billzyc) [inline react-app/src/styles/mixins.scss:1]: list of useful mixins we might want

File: react-app/src/components/googleOAuth/googleOAuth.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  const onSuccess = async (response) => {
+
+    const requestOptions = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ gToken: response.tokenId }),
+    };
+
+    const apiResponse = await fetch(
+      "http://127.0.0.1:5001/googleSignin",
```
Comment (chelseatj3) [inline react-app/src/components/googleOAuth/googleOAuth.js:21]: as we mentioned, this needs to change to :5000

File: react-app/src/styles/fonts.scss (hunk 4 above)
Comment (chelseatj3) [inline react-app/src/styles/fonts.scss:1]: At some point in time we should edit this file to contain the fonts etc that we want, although we don't need to do it now

File: react-app/src/styles/mixins.scss (hunk 7 above)
Comment (chelseatj3) [inline react-app/src/styles/mixins.scss:1]: Same with this file, not quite sure what it does now but we should edit it once we have the styles we want

**Review by chelseatj3 — APPROVED** [review]
The comments will be addressed in another pr

File: react-app/src/styles/fonts.scss (hunk 4 above)
Comment (billzyc) [reply react-app/src/styles/fonts.scss:1]: already deleted in the next pr :D

File: react-app/src/styles/mixins.scss (hunk 7 above)
Comment (billzyc) [reply react-app/src/styles/mixins.scss:1]: These are some useful mixins that we can use for styling

**Pull request merged**